public/pages/Snapshots/components/RestoreActivitiesPanel/RestoreActivitiesPanel.tsx

```diff
@@ -0,0 +1,156 @@
+import { BREADCRUMBS } from "../../../../utils/constants";
+import { ContentPanel } from "../../../../components/ContentPanel";
+
+interface RestoreActivitiesPanelProps {
+  snapshotManagementService: SnapshotManagementService;
+  snapshotId: string;
+  repository: string;
+}
+
+export const RestoreActivitiesPanel = ({ snapshotManagementService, snapshotId }: RestoreActivitiesPanelProps) => {
+  const context = useContext(CoreServicesContext);
+  const [startTime, setStartTime] = useState("");
+  const [stopTime, setStopTime] = useState("");
+  const [stage, setStage] = useState("");
+  const [indices, setIndices] = useState([{}]);
+  const [flyout, setFlyout] = useState(false);
+
+  useEffect(() => {
+    context!.chrome.setBreadcrumbs([BREADCRUMBS.SNAPSHOT_MANAGEMENT, BREADCRUMBS.SNAPSHOTS, BREADCRUMBS.SNAPSHOT_RESTORE]);
+    getRestoreStatus();
+  }, []);
+
+  const getRestoreStatus = async () => {
+    if (stage.indexOf("DONE") >= 0) {
+      console.log("done");
+      return;
+    }
+    try {
+      const res = await snapshotManagementService.getIndexRecovery();
+
+      if (res.ok) {
+        const response: GetIndexRecoveryResponse = res.response;
+
+        setRestoreStatus(response);
+        console.log(response);
+      } else {
+        context?.notifications.toasts.addDanger(res.error);
+      }
+    } catch (err) {
+      context?.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem loading the recovery."));
+    }
+  };
+
+  const onIndexesClick = (e: React.MouseEvent) => {
+    e.preventDefault();
+    setFlyout(true);
+    console.log("index clicked");
+  };
+
+  const onCloseFlyout = () => {
+    setFlyout(false);
+  };
+
+  const setRestoreStatus = (response: GetIndexRecoveryResponse) => {
+    let minStartTime: number = 0;
+    let maxStopTime: number = 0;
+    let stageIndex: number = Infinity;
+    let doneCount: number = 0;
+    const indexes: CatSnapshotIndex[] = [];
+    const stages: string[] = ["START", "INIT", "INDEX", "FINALIZE", "DONE"];
+
+    for (let item in response) {
+      if (item.indexOf("kibana") < 0) {
+        const info = response[item as keyof GetIndexRecoveryResponse].shards[0]
+        const stage = stages.indexOf(info.stage);
+        const size = `${(info.index.size.total_in_bytes / 1024 ** 2).toFixed(2)}mb`;
+
+        const time = {
+          start_time: info.start_time_in_millis,
+          stop_time: info.stop_time_in_millis,
+        };
+
+        doneCount = stage === 4 ? doneCount + 1 : doneCount;
+        stageIndex = stage < stageIndex ? stage : stageIndex;
+        minStartTime = minStartTime && minStartTime < time.start_time ? minStartTime : time.start_time;
+        maxStopTime = maxStopTime && maxStopTime > time.stop_time ? maxStopTime : time.stop_time;
+
+        if (info.source.index) {
+          indexes.push({ index: info.source.index, "store.size": size });
+        }
+      }
+    }
+    let percent = Math.floor((doneCount / indices.length) * 100);
```

Comment:
indices hasn't been set yet, should we use indexes here?

Reply:
yes, but this is early stage.  Later PR resolves this.